aptos-move/framework/aptos-framework/sources/BigVector.move

```diff
@@ -164,6 +188,29 @@ module AptosFramework::BigVector {
         }
     }
 
+    /// Reserver `additional_buckets` more buckets.
+    public fun reserve<T>(v: &mut BigVector<T>, additional_buckets: u64) {
+        while (additional_buckets > 0) {
+            Table::add(&mut v.buckets, &v.bucket_capacity, Vector::empty());
+            v.bucket_capacity = v.bucket_capacity + 1;
+            additional_buckets = additional_buckets - 1;
+        }
+    }
+
+    /// Shrink the buckets to fit the current length.
+    public fun shrink_to_fit<T>(v: &mut BigVector<T>) {
+        while (v.bucket_capacity > required_bucket(&v.length_index)) {
+            v.bucket_capacity = v.bucket_capacity - 1;
+            let v = Table::remove(&mut v.buckets, &v.bucket_capacity);
+            Vector::destroy_empty(v);
+        }
+    }
+
+    fun required_bucket(index: &BigVectorIndex): u64 {
```

Comment:
nit: `buckets_required`, as in `while (v.num_buckets > buckets_required(..)) {..}`

The name of the parameter is a bit confusing as well, because the returned number of buckets is not big enough all the time for holding `index`. How about let's call it `next_index` or `end_index` (as in the C++ [begin, end) notion, or maybe call the method `buckets_required_before(index)`